controller/manager/manager.go

```diff
@@ -261,6 +265,7 @@ func (m DefaultManager) ScaleContainer(id string, numInstances int) ScaleResult
 				errChan <- err
 				return
 			}
+			lock.Unlock()
```

Comment:
this should probably be
```
lock.Lock()
defer lock.Unlock()
```
as if there is an error the return would happen before causing a deadlock.